common/persistence/serialization/interfaces.go

```diff
@@ -35,267 +35,267 @@ import (
 		TransferProcessingQueueStates         []byte
-		TransferProcessingQueueStatesEncoding *string
+		TransferProcessingQueueStatesEncoding string
 		TimerProcessingQueueStates            []byte
-		TimerProcessingQueueStatesEncoding    *string
+		TimerProcessingQueueStatesEncoding    string
 	}
 
 	// DomainInfo blob in a serialization agnostic format
 	DomainInfo struct {
-		Name                        *string
-		Description                 *string
-		Owner                       *string
-		Status                      *int32
-		Retention                   *time.Duration
-		EmitMetric                  *bool
-		ArchivalBucket              *string
-		ArchivalStatus              *int16
-		ConfigVersion               *int64
-		NotificationVersion         *int64
-		FailoverNotificationVersion *int64
-		FailoverVersion             *int64
-		ActiveClusterName           *string
+		Name                        string
+		Description                 string
+		Owner                       string
+		Status                      int32
+		Retention                   time.Duration
+		EmitMetric                  bool
+		ArchivalBucket              string
+		ArchivalStatus              int16
+		ConfigVersion               int64
+		NotificationVersion         int64
+		FailoverNotificationVersion int64
+		FailoverVersion             int64
+		ActiveClusterName           string
 		Clusters                    []string
 		Data                        map[string]string
 		BadBinaries                 []byte
-		BadBinariesEncoding         *string
-		HistoryArchivalStatus       *int16
-		HistoryArchivalURI          *string
-		VisibilityArchivalStatus    *int16
-		VisibilityArchivalURI       *string
+		BadBinariesEncoding         string
+		HistoryArchivalStatus       int16
+		HistoryArchivalURI          string
+		VisibilityArchivalStatus    int16
+		VisibilityArchivalURI       string
 		FailoverEndTimestamp        *time.Time
-		PreviousFailoverVersion     *int64
-		LastUpdatedTimestamp        *time.Time
+		PreviousFailoverVersion     int64
+		LastUpdatedTimestamp        time.Time
 	}
 
 	// HistoryBranchRange blob in a serialization agnostic format
 	HistoryBranchRange struct {
-		BranchID    *string
-		BeginNodeID *int64
-		EndNodeID   *int64
+		BranchID    string
+		BeginNodeID int64
+		EndNodeID   int64
 	}
 
 	// HistoryTreeInfo blob in a serialization agnostic format
 	HistoryTreeInfo struct {
-		CreatedTimestamp *time.Time
+		CreatedTimestamp time.Time
 		Ancestors        []*types.HistoryBranchRange
-		Info             *string
+		Info             string
 	}
 
 	// WorkflowExecutionInfo blob in a serialization agnostic format
 	WorkflowExecutionInfo struct {
 		ParentDomainID                     UUID
-		ParentWorkflowID                   *string
+		ParentWorkflowID                   string
 		ParentRunID                        UUID
-		InitiatedID                        *int64
+		InitiatedID                        int64
 		CompletionEventBatchID             *int64
```

Comment:
This is not updated because of backward compatibility issue. Currently if this field is set to nil in thrift, it will be converted to nil in this interface, and later it will be converted to `common.EmptyEventID = -23`, which is not a zero value.
We can move the `common.EmptyEventID` logic to thrift mapper, but I don't think it's a good practice. We'll revisit it later.